hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/WebHdfsFileSystem.java

```diff
@@ -1460,6 +1472,64 @@ SnapshotDiffReport decodeResponse(Map<?, ?> json) {
     }.run();
   }
 
+  public SnapshotDiffReport getSnapshotDiffReport(final Path snapshotDir,
```

Comment:
this is by and large copied from DistributedFileSystem.getSnapshotDiffReportInternal()

Reply:
I moved common logic to DFSUtilClient.